src/utils/alloc.cpp

```diff
@@ -0,0 +1,64 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+#include <dsn/utility/alloc.h>
+
+#include <cstdlib>
+
+#include <dsn/utility/safe_strerror_posix.h>
+
+namespace dsn {
+
+#ifdef CACHELINE_SIZE
+
+/* extern */ void *cacheline_aligned_alloc(size_t size)
+{
+    if (size == 0) {
```

Comment:
> dsn_unlikely? Or dcheck_gt?

Since it's allowed for `posix_memalign` that `size` can be zero and NULL may be returned, we can use `dsn_unlikely` here.